packages/react/src/app-layout/AppLayoutSidebarNav.tsx

```diff
@@ -117,6 +140,12 @@ function AppLayoutSidebarNavListItem({
 	);
 }
 
+const isItemLink = (item: Omit<NavItem, 'label'>): item is NavLink =>
```

Comment:
This is a neat way of typeguarding, and can be found in other components (Progress Indicator, Task List, more). Not sure why we never did this in in this component.

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates